homeassistant/components/motion_blinds/cover.py

```diff
@@ -0,0 +1,278 @@
+"""Support for Motion Blinds using their WLAN API."""
+
+import logging
+
+from motionblinds import BlindType
+
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    ATTR_TILT_POSITION,
+    DEVICE_CLASS_AWNING,
+    DEVICE_CLASS_BLIND,
+    DEVICE_CLASS_CURTAIN,
+    DEVICE_CLASS_GATE,
+    DEVICE_CLASS_SHADE,
+    DEVICE_CLASS_SHUTTER,
+    CoverEntity,
+)
+
+from .const import DOMAIN, MANUFACTURER
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Motion Blind from a config entry."""
+    entities = []
+    motion_gateway = hass.data[DOMAIN][config_entry.entry_id]
+    for blind in motion_gateway.device_list.values():
+        await hass.async_add_executor_job(blind.Update)
+
+        if blind.type in [
```

Comment:
I see the "type" attribute referenced a lot.  Is this the one that was changed to "device_type" and needs to be updated or is it an attribute defined in the third-party library and therefore we have to keep it?

Reply:
No this `blind.type` is from the upstream library and comes directly from the API and is supplied by the gateway.
So yes we have to keep this one, especially since there also is a `blind.device_type` and `blind.blind_type` in the upstream library (I know it is a bit confusing but the gateway supplies diffrent kind of type information)